lib/utils/CSVDatabase.h

```diff
@@ -84,7 +85,6 @@ class CSVDatabase : public Database
      * @param[in] data The array of integer values to be written.
      * @param[in] nelements The number of integers in the array.
      */
-    virtual
```

Comment:
It looks like a lot of functions like this need `override`?

Reply:
`Database::putIntegerArray = 0` is a pure virtual function. Do we still need explicit `override`?

Reply:
The `override` may not be necessary, but my understanding is that it is used for clarity and to prevent bugs. By adding `override` to a function, the compiler will check that the function matches a virtual function. This prevents bugs like typos in the function name, different argument types (including references, pointers), etc. So it is more a matter of style and good practice than necessity.